WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/weapon/trigger/AmmoTypeSwitchTriggerListener.java

```diff
@@ -51,7 +50,7 @@ public boolean tryUse(EntityWrapper entityWrapper, String weaponTitle, ItemStack
         entityWrapper.getOffHandData().cancelTasks();
 
         Mechanics ammoTypeSwitchMechanics = getConfigurations().getObject(weaponTitle + ".Reload.Ammo.Ammo_Type_Switch.Mechanics", Mechanics.class);
-        if (ammoTypeSwitchMechanics != null) ammoTypeSwitchMechanics.use(new CastData(entityWrapper, weaponTitle, weaponStack));
+        if (ammoTypeSwitchMechanics != null) ammoTypeSwitchMechanics.use(new CastData(entityWrapper.getEntity(), weaponTitle, weaponStack));
```

Comment:
I'm seeing this a lot... maybe add constructor for `Supplier<LivingEntity>` and make the `EntityWrapper implements Supplier<LivingEntity>`